pkg/remote/resolution/resolver.go

```diff
@@ -18,11 +18,11 @@ import (
 	"errors"
 	"fmt"
 
+	remoteresource "github.com/tektoncd/pipeline/internal/resolution/resource"
```

Comment:
This is happening in a few places, but this is a breaking change to a bunch of other funcs/types because it's making input param types for other methods unreachable.

e.g. `func NewResolver(requester remoteresource.Requester, ...` needs to take in a `remoteresource.Requester`, but no external client can use that package. This makes the `Resolver` type effectively unusable for any external module.

Reply:
oh yes, you're right. 

I guess everything that depends on internal things also needs to be internal. We should probably only expose the framework package which the users need to depend on.

However, this is going to be an even bigger change than I thought and I don't know which users use the other resolution packages. 

Is resolution in beta @afrittoli @wlynch ? If so, we can still probably move things to internal and put a deprecation notice on the exposed functions. 

WDYT?

Reply:
The work here was to address [this comment thread](https://github.com/tektoncd/pipeline/pull/7845#discussion_r1578769873). I'm thinking of closing this PR and taking the approach I suggest there. Let me know what you think.